src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -2126,7 +2127,11 @@ public SecurityTokenManager getTokenManager() {
 
     @Override
     public PluginSubject getPluginSubject(Plugin plugin) {
-        return new NoopPluginSubject(threadPool);
+        Set<String> clusterActions = new HashSet<>();
+        clusterActions.add(BulkAction.NAME);
+        PluginSubject subject = new ContextProvidingPluginSubject(threadPool, settings, plugin);
+        sf.updatePluginToClusterAction(subject.getPrincipal().getName(), clusterActions);
```

Comment:
In what life-cycle stage(s) will this method will be called in?

In the current state of the implementation, this method must be called very early, as the constructed `ActionPrivileges` instance is immutable and will be only updated upon configuration change events. If this method gets called at the wrong time, changes won't be picked up immediately.